### ISSUE TITLE
Still interested in working on this?

### ISSUE THREAD

**Issue by richardklafter** [kipraske/web-brogue-client#2]
I got brogue working at a reasonable frame rate using emscripten. I put a temp demo site up at http://icyego.com:8080/. Mouse input and save games are not implemented yet. I will get things cleaned up a little and on GitHub in the next day or so.

I only found your project after I had it working so it uses a canvas.

**Comment by kipraske** [comment]
Yes! I certainly am. Unfortunately, I probably going to be too busy to start working with you until October, but I will be very excited until then. This project was just to save my research into trying to archive what you have now done. I couldn't get the libtcod part of the brogue source code to compile with emscripten, this was more of a proof of concept starting point.

I am very excited because this is the missing link to setting up a better web brogue (London Server: http://brogue.roguelikelike.com/ and My server: brogue.lumenstone.org). Currently that code base is being maintained by Flend at https://github.com/flend/web-brogue.

The point of wanting to create a brogue-client with emscripten is to create a robust frontend client for my web server. Since brogue is completely deterministic we can have the game run on the client and on the server and not have to worry about sending the huge amount of cell update data across the net, we would just need to send what our key updates were and play both the client and the server with the same seed.

Secondly, the web brogue server currently runs all of the games in memory forever. This was Flend's solution to the fact that if the Node.js server crashes (which it does from time to time), all of these games will still be available. This works very well but we still end up losing everyones game if the server machine needs to be reset, and two it is not very scalable -- since we would eventually run out of memory for all of the brogue processes. We only have about 20 active users at any given time so really this plan has been working pretty well despite my misgivings. My whole point though is that if we have a client-server game pair going on we could keep the state on the client and then when the server comes back up we can update the server accordingly, and thus we could just go back to using brogue save files again which would be able to persist across server resets.

**Comment by richardklafter** [comment]
I cleaned things up a bit and published what I have so far to GitHub (mouse
now works). https://thingless.github.io/broguejs/.

I did not end up compiling libtcod portion of brogue (because emscripten's
docs suck). I wrote my own platform file instead.

When ya get around to it I would be glad to help ya access emscripten's
local file system to get saved games and high scores (I have a branch that
saves them to the browser).

Best,
Richard
